File: backend/src/main/java/dev/mikeqwe/stackexchange/piano/task/backend/domain/Question.java
package dev.mikeqwe.stackexchange.piano.task.backend.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Question {
    private List<String> tags;
    private ShallowUser owner;
    private Boolean isAnswered;
    private Integer viewCount;
    private Integer upVoteCount;
    private Integer answerCount;
    private Integer score;
    private Instant creationDate;
    private String link;
    private String title;
}

File: frontend/src/app/entity/question.ts
export class Question {
  title: string;
  tags: string[];
  owner: ShallowUser;
  isAnswered: boolean;
  viewCount: number;
  upVoteCount: number;
  answerCount: number;
  score: number;
  creationDate: Date;
  link: string;
}

export class ShallowUser {
  reputation: number;
  displayName: string;
}
